### PR TITLE
KTOR-5543 KTOR-5544 Update Kotlin to 1.8

### PR DESCRIPTION
**Pull request by e5l**
Fix [KTOR-5543](https://youtrack.jetbrains.com/issue/KTOR-5543)
Fix [KTOR-5544](https://youtrack.jetbrains.com/issue/KTOR-5544)
Fix [KTOR-5698](https://youtrack.jetbrains.com/issue/KTOR-5698)

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -26,13 +22,21 @@ org.gradle.jvmargs=-Xmx6g -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8
...
+kotlin.native.ignoreIncorrectDependencies=true
+kotlin.native.binary.memoryModel=experimental
+kotlinx.atomicfu.enableJvmIrTransformation=true
+kotlinx.atomicfu.enableJsIrTransformation=true
 
-kotlin_version=1.7.20
+kotlin_version=1.8.10
 coroutines_version=1.6.4
-atomicfu_version=0.18.5
+atomicfu_version=0.18.0
```
[the remark this answers is not in the document]
Comment (e5l) [reply gradle.properties:36]: thanks

**Comment by marychatte** [comment]
There is `Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.` exception in `Java 8`. And build failure in `JavaScript on Chrome/Node.js` seems to be related too

**Comment by e5l** [comment]
Thanks, will check

**Review by marychatte — COMMENTED** [review]
`Jte tests` are failing with `Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.8.0, expected version is 1.6.0.` exception in Java 8. And `Task 'cleanJsIrTest' not found in root project 'ktor'` in JavaScript on Chrome/Node.js

File: build.gradle.kts [hunk 2]
```diff
@@ -93,7 +93,7 @@ apply(from = "gradle/compatibility.gradle")
 plugins {
     id("org.jetbrains.dokka") version "1.7.20" apply false
     id("org.jetbrains.kotlinx.binary-compatibility-validator") version "0.12.1"
-    id("kotlinx-atomicfu") version "0.18.5" apply false
+    id("kotlinx-atomicfu") version "0.19.0" apply false
```
Comment (marychatte) [inline build.gradle.kts:96]: Can we use the latest `0.20.0` version?

File: build.gradle.kts [hunk 3]
```diff
@@ -227,3 +226,22 @@ fun KotlinMultiplatformExtension.configureSourceSets() {
...
+
+    val jsIrTest by tasks.creating {
+        dependsOn(jsTest)
+    }
+}
+
+val jsLegacyTest by tasks.creating {
+}
+
+val cleanJsLegacyTest by tasks.creating {
```
Comment (marychatte) [inline build.gradle.kts:246]: Why do we need those?

**Review by marychatte — APPROVED** [review]

**Pull request merged**